### PR TITLE
Activity Indicator

### LINKED ISSUE #101: Add ActivityIndicator component
We need an activity indicator component for the HN app:

https://developer.apple.com/design/human-interface-guidelines/macos/indicators/progress-indicators/

This task also includes integrating the activity indicator into the main and detail screens of the HN app.

### PR DESCRIPTION
**Pull request by wokalski**
Fixes #101 

image

### CODE REVIEW

File: examples/hacker-news/app/StoryList.re [hunk 1]
```diff
@@ -113,12 +113,16 @@ let component = {
...
       (
         hooks,
-        <scrollView
-          onReachedEnd=loadNextPage style=[flex(1.), backgroundColor]>
-          {switch (stories) {
-           | Loading => <activityIndicator style=[flex(1.)] />
-           | stories =>
-             stories
+        switch (stories) {
+        | Loading =>
```
Comment (rauanmayemir) [inline examples/hacker-news/app/StoryList.re]: This only works once for the initial load. We might want to add another kind of indicator for 'pull-to-refresh', file an issue for that?

File: renderer-macos/lib/stubs/BriskProgressIndicator.c [hunk 2]
```diff
@@ -2,8 +2,17 @@
...
   NSProgressIndicator *progressIndicator = [NSProgressIndicator new];
+  progressIndicator.style = NSProgressIndicatorStyleSpinning;
   [progressIndicator startAnimation:nil];
   progressIndicator.indeterminate = YES;
   retainView(progressIndicator);
   return progressIndicator;
 }
+
+double ml_BriskProgressIndicator_fittingWidth(NSProgressIndicator *indicator) {
+  return indicator.fittingSize.width;
```
Comment (rauanmayemir) [inline renderer-macos/lib/stubs/BriskProgressIndicator.c]: You're setting `ml_BriskProgressIndicator_fittingWidth_bc` in the stubs, but there's no impl for it. 🤷‍♂️

Comment (wokalski) [reply renderer-macos/lib/stubs/BriskProgressIndicator.c]: I know! That's on purpose. If someone ever needs to use byte code versions they'll see the problem in the compile time.

File: examples/hacker-news/app/StoryList.re (hunk 1 above)
Comment (wokalski) [reply examples/hacker-news/app/StoryList.re]: yes, sure!

File: renderer-macos/lib/stubs/BriskProgressIndicator.c (hunk 2 above)
Comment (rauanmayemir) [reply renderer-macos/lib/stubs/BriskProgressIndicator.c]: Neat. 👌

**Pull request merged**